Task2Logic/Hex.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2Logic
+{
+    public static class Hex
+    {
+        /// <summary>
+        /// convert int digit to hex digit
+        /// </summary>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        private static char HexDigit(int num)
+        {
+            return num < 10 ? (char)(num + 48) : (char)(num + 55);
+        }
+        /// <summary>
+        /// convert hex digit to int digit
+        /// </summary>
+        /// <param name="val"></param>
+        /// <returns></returns>
+        public static int ConvertHexDigit(char val)
+        {
+            if ((int)val <= 57 && (int)val >= 48)
+                return (int)val - 48;
```

Comment:
You use magic numbers!!! IT'S BAD :(
Instead it's better to use something like 
byte aIndex = (byte)'a'

Btw in case of char it's better to compare chars with chars like if ( 'a' <= val && val <= 'f') { ...